common-app-lib/src/cmr/common_app/api/log.clj

```diff
@@ -0,0 +1,133 @@
+                      (with-out-str (pp/pprint data)))
+        content (if html?
+                  {cr/CONTENT_TYPE_HEADER mt/html}
+                  {cr/CONTENT_TYPE_HEADER mt/edn})]
+    {:status status-code
+     :body body-string
+     :headers content}))
+
+(defn get-log-error-response
+  "Creates an error response if the content type is not correctly set. If it is not correctly set
+   or not set at all, then the compojure wrap-params will consume the body and it will be blank.
+   Therefore I will send back an error to correctly set the content-type. Otherwise I will have to
+   save the body in another key (like search is doing) in all services and use that key instead."
+  [status-code content-type]
+  {:status status-code
+   :body (str "The content type of "
+              content-type
+              " in the request is either not provided or is not of the correct type. Please set the request Content-Type to application/edn.")
+   :headers {cr/CONTENT_TYPE_HEADER "text/plain"}})
+
+(defn get-logging-configuration
+  "This function gets the logging configuration map to the caller. If a token is not passed in
+   with the request or the user doesn't have permission, the acl will through an exception. The
+   exception is caught in base code and an appropriate error message is passed to the end user."
+  [request-context headers]
+  (acl/verify-ingest-management-permission request-context :update)
+  (get-log-configuration-response
+    (.contains (mt/get-header headers "accept") "text/html")
+    200
+    (log/get-logging-configuration)))
+
+(defn merge-log-configuration
+  "This function asks the logger to merge the passed in configuration with the current logging
+   configuration. If a token is not passed in with the request or the user doesn't have permission,
+   the acl will through an exception. The exception is caught in base code and an appropriate error
+   message is passed to the end user."
+  [request-context headers body]
+  (acl/verify-ingest-management-permission request-context :update)
+  (let [body-string (slurp body)]
+    ;; This tests to see if the content type is correct. If the user didn't set the correct
+    ;; content type or didn't set it at all then the compojure wrap-params will consume the
+    ;; body and it will be blank when the body gets here.
+    (if (string/blank? body-string)
+      (get-log-error-response 400 (mt/get-header headers "Content-Type"))
+      ;; Put the string into an edn format.
+      (let [edn-map-config (edn/read-string body-string)]
+        (log/warn (format "Merging a new CMR logging configuration map %s. The old configuration is %s."
+                          edn-map-config
+                          (with-out-str (pp/pprint (log/get-logging-configuration)))))
+        (get-log-configuration-response
+          (.contains (mt/get-header headers "accept") "text/html")
+          200
+          (log/merge-logging-configuration edn-map-config))))))
+
+(def log-api-routes
```

Comment:
I'd probably just name this `logging-routes`, since it's not doing anything (it's just a definition).

Reply:
done.